src/Parser.php

```diff
@@ -302,11 +299,8 @@ protected function parseBackground()
         $line = $token['line'];
 
         if (count($this->popTags()) !== 0) {
-            throw new ParserException(sprintf(
-                'Background can not be tagged, but it is on line: %d%s',
-                $line,
-                $this->file ? ' in file: ' . $this->file : ''
-            ));
+            // Should not be possible to happen, parseTags should have already picked this up.
+            throw new UnexpectedTaggedNodeException($token, $this->file);
```

Comment:
I have kept this for the moment out of caution, I don't think it can happen now (it doesn't with any of the test examples).